### PR TITLE
Expanded options for multirests layout

### LINKED ISSUE #23918: "1" above the one-measure rest in parts
## My idea

I suggest adding the ability for users to mark one-measure rests with the number "1" in instrumental parts.

## Problem to be solved

This will allow the performer to be sure that the rest is not multi-measure and that nothing was missed. 
At the moment, in MuseScore Studio, the number "1" can only be specified manually, while aligning it in the center above the bar. 
Снимок экрана (472)

If you try to specify a minimum value of "1" for multi-measure rests in the style settings, this will not be displayed correctly.
Снимок экрана (474)

## Additional context

The number "1" should not be specified if the bar is **_G.P._** or there is a cue in it (but see #23912)

## Checklist

- ✅ This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- ✅ I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

### LINKED ISSUE #24945: Multimeasure rests on multistaff parts
### Your idea

For a part that contains multiple staves, such as a keyboard or multi-instrument percussion part or even a multiplayer part such as Flute 1 & 2, multimeasure rest number should appear centered between staves and not on the top of each staff.

### Problem to be solved

Currently, MuseScore displays the number at the top of each staff like this:

Screen Shot 2024-09-27 at 7 35 08 AM

But more standard engraving practice is for them to appear between staves like this:

Screen Shot 2024-09-27 at 7 37 03 AM

### Prior art

Dorico does this automatically. For Finale, it is possible by running a plugin I wrote.

### Additional context

_No response_

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

### LINKED ISSUE #24947: Additional options for single bar "multimeasure" rests
### Your idea

While researching #23918 I realized that whereas the requested feature in that issue already exists, there are a couple of other nice-to-have options with regards to multimeasure rests that encompass a single measure.

- the ability of a single bar "multimeasure" rest automatically to have the same width as an empty bar.
- automatically suppress the printing of a multimeasure rest range under a single-bar multimeasure rest, and instead display the measure number normally (if it would be displayed otherwise).

Beyond this, other settings that would be nice are:

- for old symbols, a different spacing value when there are 3 or more symbols so that they can be more closely spaced.

### Problem to be solved

Currently, both the spacing and measure number positioning of a single-bar multimeasure rest can appear awkward if

- old-style symbols are being used
- measure ranges beneath mm rests is selected

### Prior art

I have not researched what Dorico does. These were options in my Multimeasure Rest plugin for Finale.

### Additional context

I am opening this issue separately so that #23918 can be closed. If this is a desirable feature I'm willing to work on it.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #23918 
Resolves: #24947 
Resolves: #24945 

This is based on #25099, so that should be merged first.

- New option for constant-width multirests (where constant means independent of the time signature). 
Normal:
Screenshot 2024-10-17 142357
Constant:
Screenshot 2024-10-17 142340
- New option for displaying multirest number in-between staves for multi-stave instruments. This is now the default.
Before:
Screenshot 2024-10-17 142447
After:
Screenshot 2024-10-17 142439
- New options for the specific case of 1-bar multirest.
- The new default for "Minimum number of empty measures" has been changed to 1.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/engraving/rw/read410/tread.cpp [hunk 1]
```diff
@@ -3436,6 +3436,10 @@ void TRead::read(MMRest* r, XmlReader& e, ReadContext& ctx)
             NoteDot* dot = Factory::createNoteDot(r);
             TRead::read(dot, e, ctx);
             r->add(dot);
+        } else if (tag == "mmRestNumberPos") {
+            // Old property, deprecated in 4.5
+            r->setNumberOffset(e.readDouble() - ctx.style().styleS(Sid::mmRestNumberPos).val());
+        } else if (tag == "mmRestNumberPos") {
```
Comment (miiizen) [inline src/engraving/rw/read410/tread.cpp:3442]: This one looks unneeded - did you mean
`else if (TRead::readProperty(r, tag, e, ctx, Pid::MMREST_NUMBER_OFFSET))`?

File: src/engraving/rw/read400/tread.cpp [hunk 2]
```diff
@@ -3051,6 +3051,10 @@ void TRead::read(MMRest* r, XmlReader& e, ReadContext& ctx)
             NoteDot* dot = Factory::createNoteDot(r);
             TRead::read(dot, e, ctx);
             r->add(dot);
+        } else if (tag == "mmRestNumberPos") {
+            // Old property, deprecated in 4.5
+            r->setNumberOffset(e.readDouble() - ctx.style().styleS(Sid::mmRestNumberPos).val());
+        } else if (TRead::readProperty(r, tag, e, ctx, Pid::MMREST_NUMBER_OFFSET)) {
```
Comment (miiizen) [inline src/engraving/rw/read400/tread.cpp:3057]: I don't think we need this. The property is introduced in 4.5 and shouldn't show up in in pre-4.5 scores

Comment (mike-spa) [reply src/engraving/rw/read400/tread.cpp:3057]: Well spotted, thanks, I obvs meant to put this in the 410 reader

**Review by miiizen — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
> * New options for the specific case of 1-bar multirest.

This is fine

> * The new default for "Minimum number of empty measures" has been changed to 1.

But this IMHO is not. See also discussion in https://musescore.org/en/node/393912

**Pull request merged**